Lib/test/test_email/test_email.py

```diff
@@ -3032,6 +3032,20 @@ def test_parseaddr_empty(self):
         self.assertEqual(utils.parseaddr('<>'), ('', ''))
         self.assertEqual(utils.formataddr(utils.parseaddr('<>')), '')
 
+    def test_parseaddr_multiple_domains(self):
+        self.assertEqual(
+            utils.parseaddr('a@b@c'),
+            ('', 'a@')
```

Comment:
According to the docs:
> email.utils.parseaddr(address)
>    Parse address – which should be the value of some address-containing field such as To or Cc – into its constituent realname and email address parts. Returns a tuple of that information, unless the parse fails, in which case a 2-tuple of ('', '') is returned.

Can we expect the same in this case?

I made this change to make that happen:
```
@@ -379,7 +379,12 @@ class AddrlistClass:
         aslist.append('@')
         self.pos += 1
         self.gotonext()
-        return EMPTYSTRING.join(aslist) + self.getdomain()
+        domain = self.getdomain()
+        if not domain:
+            # Invalid domain, return an empty address instead of returning a
+            # local part to denote failed parsing.
+            return EMPTYSTRING
+        return EMPTYSTRING.join(aslist) + domain
```

Reply:
Sure, please confirm that I got it right:

You mean that `utils.parseaddr('a@b@c')` should return `('', '')`, or just `''` ?

Just one question about consistency, would that change also implicate a change of the following : 

```
>>> parseaddr('foo@a.') 
('', 'foo@a.')
```

If so, should that also be in the security patch or do you want a new issue and a new patch for breaking backward compatibility ?

Thanks for your review !

Reply:
Yes, I meant return value should be `('', '')`.

Coming to the behavior of `parseaddr`, here is my thinking on this this issue. Right now, we return either "whatever we can parse" (which is why we return the invalid domains as-is without opinions) and `('', '')` when there is an exception raised during parsing.

So, following this line of reasoning, with domains like `a@b@c.com` should actually return `('', ,'a@b@c.com' )`. I think this is the right thing to do. *If* you think this behavior can cause security issue for the downstreams who might have similar bug and would take in `a@b@c.com` as `a@b` and think we should consider this, then the next right thing to do would be to return an empty string denoting unparseable address.

Does that line of reasoning made like it did in my head?

Reply:
Yes of course it is expected to break BC to fix security issue. In this case do you want this patch to also change parseaddr('a@b.') to return a tuple of two empty strings in this patch for consistency with parseaddr('a@b@c') which is now going to return an empty strings tuple ? Sorry you make sense but there i am not sure if this consistency point is important or not for the time being

Reply:
I think we can deal with this later, so I added your patch, thanks @maxking ! But I didn't know how to add credit

Reply:
You can add credits in commit message, I think.

You may need to update your tests to reflect the patch too, they seem to be failing right now with new updates.